src/ios/CDVOrientation.m

```diff
@@ -80,7 +80,13 @@ -(void)screenOrientation:(CDVInvokedUrlCommand *)command
             }
             if (value != nil) {
                 _isLocked = true;
-                [[UIDevice currentDevice] setValue:value forKey:@"orientation"];
+                if (@available(iOS 16.0, *)) {
+                    #if __IPHONE_OS_VERSION_MAX_ALLOWED > __IPHONE_15_5 // Xcode 14 and iOS 16, or greater
+                        [self.viewController setNeedsUpdateOfSupportedInterfaceOrientations];
+                    #endif
+                } else {
+                    [UINavigationController attemptRotationToDeviceOrientation];
```

Comment:
`[[UIDevice currentDevice] setValue:value forKey:@"orientation"];`

Reply:
Do you want me to update line 88 to that?

Reply:
According to the [Apple Docs](https://developer.apple.com/documentation/uikit/uidevice/1620053-orientation?language=objc), the `orientation` property of `UIDevice` instance is read-only, so I assume setting a value for it is a no-op.

Reply:
Not sure, but if you have any suggestions or fixes, feel free to submit them on my batch branch.

I'll definitely get them merged https://github.com/jaydrogers/cordova-plugin-screen-orientation/tree/patch-1

Sorry I can't be much help on this other than the dude who opened the PR 😆

Reply:
Hello and thanks for the fork :)
the change suggested by msmtamburro on line 88 is indeed important because we need to set value set in javascript and not come back to default device orientation (which is the default orientation calculate by device giroscope and used before in the code (line 75-79) when "unlock" is called (orientationMask==15))